### PR TITLE
feat(fs): add the `size` method to get the size of a file or directory

### PR DESCRIPTION
**Pull request by ayangweb**
@FabianLars Help me refine it a bit more, thanks!

### CODE REVIEW

File: plugins/fs/guest-js/index.ts [hunk 1]
```diff
@@ -1322,6 +1322,28 @@ async function watchImmediate(
...
+ * Get the size of a file or directory.
+ * @example
+ * ```typescript
+ * import { size, BaseDirectory } from '@tauri-apps/plugin-fs';
+ * // Get the size of the `$APPDATA/tauri` directory.
+ * const dirSize = await size('tauri', { baseDir: BaseDirectory.AppData });
+ * console.log(dirSize); // 1024
+ * ```
+ *
+ * @since 2.0.0
```
Comment (FabianLars) [inline plugins/fs/guest-js/index.ts]:
```suggestion
 * @since 2.1.0
```

File: plugins/fs/src/commands.rs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 // SPDX-License-Identifier: MIT
 
+use anyhow::Ok;
```
Comment (FabianLars) [inline plugins/fs/src/commands.rs]:
```suggestion
```

Do we really need this? If so, which line does rust complain about if it's not there?

File: plugins/fs/guest-js/index.ts [hunk 3]
```diff
@@ -1322,6 +1322,28 @@ async function watchImmediate(
   }
 }
 
+/**
+ * Get the size of a file or directory.
```
Comment (FabianLars) [inline plugins/fs/guest-js/index.ts]:
```suggestion
 * Get the size of a file or directory. For files, the `stat` functions can be used as well.
 *
 * If `path` is a directory, this function will recursively iterate over every file and every directory inside of `path` and therefore will be very time consuming if used on larger directories.
```

Maybe we should also add a note on the stat functions that it does not return the expected dir size and that this function should be used if that is required.

**Review by FabianLars — CHANGES_REQUESTED** [review]
thanks! just a few small thinks. Also, we'll need a changefile like https://github.com/tauri-apps/plugins-workspace/blob/v2/.changes/fix-fs-scope-escape-paths.md

File: plugins/fs/src/commands.rs (hunk 2 above)
Comment (ayangweb) [reply plugins/fs/src/commands.rs]: Haha, I didn't even notice using it

**Comment by ayangweb** [comment]
Thank you for your guidance, I've revised it, and you can look at it again!

File: .changes/add-fs-size.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+fs: minor
+persisted-scope: minor
+---
+
+**Breaking Change:** add the `size` method to get the size of a file or directory.
```
Comment (FabianLars) [inline .changes/add-fs-size.md]:
```suggestion
Add the `size` method to get the size of a file or directory.
```
My bad, i gave you a stupid example file. This change is not breaking.

File: .changes/add-fs-size.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
+---
+fs: minor
+persisted-scope: minor
```
Comment (FabianLars) [inline .changes/add-fs-size.md]:
```suggestion
```

Same

**Review by FabianLars — COMMENTED** [review]
I'll review the rest later, I'm just on a phone rn

**Comment by ayangweb** [comment]
```md
---
fs: minor
---

Add the `size` method to get the size of a file or directory.
```

Change it to this?

File: plugins/fs/guest-js/index.ts [hunk 6]
```diff
@@ -1322,6 +1322,30 @@ async function watchImmediate(
   }
 }
 
+/**
+ * Get the size of a file or directory. For files, the `stat` functions can be used as well.
+ *
+ * If `path` is a directory, this function will recursively iterate over every file and every directory inside of `path` and therefore will be very time consuming if used on larger directories.
+ * @example
```
Comment (FabianLars) [inline plugins/fs/guest-js/index.ts:1330]:
```suggestion
 *
 * @example
```

**Comment by FabianLars** [comment]
the errors should be gone once you add the command to the invoke_handler list https://github.com/ayangweb/plugins-workspace/blob/fs-size/plugins/fs/src/lib.rs#L396

**Comment by ayangweb** [comment]
> the errors should be gone once you add the command to the invoke_handler list https://github.com/ayangweb/plugins-workspace/blob/fs-size/plugins/fs/src/lib.rs#L396

ok, I added it!

**Review by amrbashir — DISMISSED** [review]

**Comment by amrbashir** [comment]
looks like you also need to run `pnpm build` and potentially `pnpm format`

**Comment by ayangweb** [comment]
> looks like you also need to run `pnpm build` and potentially `pnpm format`

Okay, I executed and submitted it!

**Review by amrbashir — APPROVED** [review]

**Review by FabianLars — APPROVED** [review]

**Pull request merged**